### ISSUE TITLE
doc scope requirements for parallelism

### ISSUE THREAD

**Issue by bob-carpenter** [stan-dev/rstan#260]
The real issue behind the closed issue #257 is that there needs to be doc on the initialization functions for `stan()` and related functions that indicate the scope requirements on functions.  I'm not sure what they are, or I'd list them!

@bgoodri commented https://github.com/stan-dev/rstan/issues/257#issuecomment-178158365, which I repeat here:

I think this

```
Error in init() : object 'ThetaInit' not found

error in specifying arguments; sampling not done
```

occurs because `ThetaInit` is an object in the `GlobalEnv` of the parent R process and cannot be accessed via the usual scoping mechanisms by any of the child R processes. In other words, the init function is exported to the child R processes so that it can be called but it can only refer to `chain_id` because everything else in the parent process is out of scope.

**Comment by bgoodri** [comment]
Or maybe they can be evaluated in the parent process before exporting them to the child process.

**Comment by bgoodri** [comment]
@maverickg Do I need to do any more than d47d6ba5c4f9cb619a2d8bd3e48106c3f86252a9 to have the init function method work in parallel when the function refers to an object in the global environment of the parent R process?

**Comment by bgoodri** [comment]
Actually, 71bdd1a is needed as well. Also, do you know what is wrong with Travis?

**Comment by maverickg** [comment]
@bgoodri thanks

I think those should be enough.  I am not going to remove some redundant code in function config_argss in misc.R (I suppose there is no way that init is a function at that point).

**Comment by bgoodri** [comment]
Okay. I was just about to upload a new version to CRAN. Is the redundant
code harmful?

**Comment by maverickg** [comment]
no.

**Comment by maverickg** [comment]
by the way, I don't know the issues with travis.  Neither Linux nor Mac works for different reasons.
